packages/block-library/src/image/image.js

```diff
@@ -83,9 +83,29 @@ const scaleOptions = [
 	},
 ];
 
-const disabledClickProps = {
-	onClick: ( event ) => event.preventDefault(),
-	'aria-disabled': true,
+// If the image has a href, wrap in an <a /> tag to trigger any inherited link element styles.
+const ImageWrapper = ( { href, children } ) => {
+	if ( ! href ) {
+		return children;
+	}
+	return (
+		<a
+			href={ href }
+			onClick={ ( event ) => event.preventDefault() }
+			aria-disabled={ true }
+			style={ {
+				// When the Image block is linked,
+				// it's wrapped with a disabled <a /> tag.
+				// Restore cursor style so it doesn't appear 'clickable'
+				// and remove pointer events. Safari needs the display property.
+				pointerEvents: 'none',
+				cursor: 'default',
```

Comment:
@t-hamano Do you think this could be confusing for low vision users to get a normal mouse pointer when the link is not clickable?

Reply:
On the face of it, `default` seems appropriate to me given that the link is disabled. (?)

`cursor: 'pointer'` is the "hand", or default browser cursor, for linked elements.

Reply:
I think that since this element cannot actually be clicked, it is okay to leave the cursor as the default.

By the way, #56123 has been submitted to add a pressed state to the link button in the block toolbar to make it easier to tell if the image is _linked_.